src/com/zalinius/zje/music/synths/SoundFactory.java

```diff
@@ -0,0 +1,185 @@
+	public static final Buffer RESONANT_STRONG = makeBuffer(sinePower(20),"Resonant Ultra");
+	public static final Buffer RESONANT_SHARP = Buffer.TRIANGLE;
+
+	public static final Buffer STRING_SOFT = makeBuffer(normalizer(test()),"Soft String");
+	public static final Buffer STRING_STRONG = makeBuffer(normalizer(sinEnvelope()),"Deep String");
+	public static final Buffer STRING_SHARP = makeBuffer(squareSine(),"Sharp String");
+	
+	public static Buffer makeBuffer(WaveFunction function, String name) {
+		BufferFactory factory = new BufferFactory() {
+			
+			@Override
+			public String getName() {
+				return name;
+			}
+			
+			@Override
+			public Buffer generateBuffer(int bufferSize) {
+		    	Buffer b = new Buffer(bufferSize);
+		        for(int i = 0; i < bufferSize; i++) {
+		        	b.buf[i] = function.value(i, bufferSize);
+		        }
+		    	return b;
+			}
+		};
+		
+		return factory.getDefault();
+	}
+
+	
+	private static WaveFunctionFractional sinEnvelope() {
+		return new WaveFunctionFractional() {
+			
+			@Override
+			public double value(double fraction) {
+				return Math.sin(fraction/2)*Math.pow(Math.cos(2*fraction),7);
+			}
+		};
+	}
+	
+	private static WaveFunctionFractional test() {
+		return new WaveFunctionFractional() {
+			
+			@Override
+			public double value(double fraction) {
+				return Math.sin(fraction/2) * Math.sin(2*fraction);//TODO quite a nice sound: 
+			}
+		};
+	}
+	
+	private static WaveFunction squareSine() {
+		return new WaveFunction() {
+
+			@Override
+			public float value(int i, int max) {
+				return (Buffer.SINE.getValueIndex(3*i) *Buffer.SQUARE.getValueIndex(i));
+			}
+
+		};
+	}
+
+	public static WaveFunction absoluteSine() {
+		return new WaveFunction() {
+			
+			@Override
+			public float value(int i, int max) {
+				return (float)Math.sin(Math.PI * (double)i / (double)max);
+			}
+		};
+	}
+
+	public static WaveFunction halfSaw() {
+		return new WaveFunction() {
+			
+			@Override
+			public float value(int i, int max) {
+				return Buffer.SAW.getValueIndex(i)/4;
+			}
+		};
+	}
+	public static WaveFunction halfSquare() {
+		return new WaveFunction() {
+			
+			@Override
+			public float value(int i, int max) {
+				return Buffer.SQUARE.getValueIndex(i)/4;
+			}
+		};
+	}
+	public static WaveFunction sinePower(double power) {
+		if(power < 1) {
+			power = 1;
+		}
+		final double trueLevel=power;
+		return new WaveFunction() {
+			
+			@Override
+			public float value(int i, int max) {
+				double sineValue = Buffer.SINE.getValueIndex(i);
+				if(sineValue > 0d) {
+					return (float) Math.pow(sineValue,trueLevel);
+				}
+				else {
+					return (float) -Math.pow(Math.abs(sineValue),trueLevel);
+				}
+			}
+		};
+	}
+	
+	public static WaveFunction normalizer(WaveFunctionFractional other) {
+		return new WaveFunction() {
+			
+			@Override
+			public float value(int i, int max) {
+				return (float) other.value(Math.PI * 2*(double) i / (double)max);
+			}
+		};
+	}
+
+}
```

Comment:
Lotta raw numbers in this file, though I'm not sure I know enough about beads whether they're ok or not

Reply:
Some of it should have stayed in PolygonPal, and I moved it back, some of it I gave new Constant names to, and some of them are basically magic numbers because they sound good :P